File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s= s.toLowerCase();
        String str="", rev="";
        for(int i=0; i<s.length(); i++)
        {
            char ch= s.charAt(i);
            if ( Character.isLetter(ch) || Character.isDigit(ch) )
            str= str + ch;
        }
        for(int i=str.length()-1; i>=0; i--)
        {
            rev= rev+ str.charAt(i);
        }
        if ( rev.equals(str) )
        return true;
        else
        return false;
    }
}

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        l=0
        r= len(s)-1 
        s=s.lower()

        while (l<=r):
            if not (s[l].isalpha() or s[l].isdigit()):
                l+=1
                continue
            if not (s[r].isalpha() or s[r].isdigit()):
                r-=1  
                continue
            if s[l]!=s[r]:
                return False 
            else:
                l+=1
                r-=1
        return True
                